core/ts/Poll.ts

```diff
@@ -161,53 +157,41 @@ export class Poll implements IPoll {
     this.stateTree = this.maciStateRef.stateTree.copy();
 
     // Create as many ballots as state leaves
-    const emptyBallot = new Ballot(this.maxValues.maxVoteOptions, this.treeDepths.voteOptionTreeDepth);
-    const emptyBallotHash = emptyBallot.hash();
-    this.ballotTree = new IncrementalQuinTree(this.stateTreeDepth, emptyBallot.hash(), STATE_TREE_ARITY, hash5);
-    this.ballotTree.insert(emptyBallotHash);
+    this.emptyBallotHash = this.emptyBallot.hash();
+    this.ballotTree = new IncrementalQuinTree(this.stateTreeDepth, this.emptyBallotHash, STATE_TREE_ARITY, hash5);
+    this.ballotTree.insert(this.emptyBallot.hash());
 
+    // we fill the ballotTree with empty ballots hashes to match the number of signups in the tree
     while (this.ballots.length < this.stateLeaves.length) {
-      this.ballotTree.insert(emptyBallotHash);
-      this.ballots.push(emptyBallot);
+      this.ballotTree.insert(this.emptyBallotHash);
+      this.ballots.push(this.emptyBallot);
     }
 
-    this.numSignUps = Number(this.maciStateRef.numSignUps.toString());
-
     this.stateCopied = true;
   };
 
   /**
    * Process one message at index `index`.
```

Comment:
It accept `message` and `encPubKey` as parameters now, so the description here should be modified.